src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -899,6 +1042,212 @@ client_session* get_session(document::view op, entity::map& map) {
     return &map.get_client_session(session_name);
 }
 
+document::value run_command(database& db, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::view command = arguments["command"].get_document().value;
+
+    return db.run_command(command);
+}
+
+document::value update_one(collection& coll, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::view filter = arguments["filter"].get_document().value;
+    document::view update = arguments["update"].get_document().value;
+    options::update options{};
+
+    if (arguments["collation"]) {
+        options.collation(arguments["collation"].get_document().value);
+    }
+    if (arguments["hint"]) {
+        if (arguments["hint"].type() == bsoncxx::v_noabi::type::k_string)
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+    if (arguments["upsert"]) {
+        options.upsert(arguments["upsert"].get_bool().value);
+    }
+
+    std::int32_t matched_count = 0;
+    bsoncxx::stdx::optional<std::int32_t> modified_count;
+    bsoncxx::stdx::optional<types::bson_value::view> upserted_id{};
+
+    auto update_one_result = coll.update_one(filter, update, options);
+    if (update_one_result) {
+        matched_count = update_one_result->matched_count();
+
+        // Server versions below 2.4 do not return an `nModified` count.
```

Comment:
I'm not sure about this either. But as of C++ driver 3.3, only 3.0+ is supported: https://docs.mongodb.com/drivers/cxx/#mongodb-compatibility). I suggest removing the `try/catch`. If tests do fail, I bet it is the same culprit (unacknowledged writes).

Reply:
Removed the `try/catch` and tests seem to pass!